ProjFS.Mac/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -573,13 +575,10 @@ errno_t ActiveProvider_SendMessage(VirtualizationRootHandle rootIndex, const Mes
     
     if (nullptr != userClient)
     {
-        uint32_t messageSize = sizeof(*message.messageHeader) + message.messageHeader->pathSizeBytes;
```

Comment:
As you've nicely done, seperating the size / encoding into separate functions requires the need to do several checks for sizing (the functions are independent now).

The advantage would be we can unit test.  I *think* this is not being picked up by code coverage because this class isn't part of the unit tests project.  I.e. we mock out sending the request.

Would it make sense to add the new functions to unit tests?  Does anyone else have feelings on the approach? @wilbaker

Reply:
I think we should debate unit testing once we have a fundamental approach we can agree on. But yeah, unit testing serialisation/deserialisation code is definitely a good idea I think.